.github/workflows/tests.yml

```
@@ -5,12 +5,56 @@ on:
   # but only for the main branch
   push:
     branches:
-      - main  # Trigger on push events to the main branch
+      - main # Trigger on push events to the main branch
       - beta # Trigger on push events to the beta branch
+    # go upvote https://github.com/actions/runner/issues/1182 and yell at microsoft until they fix this or ditch yml for workflows
+    paths:
+      # src and test files
+      - "src/**"
+      - "test/**"
+      - "public/**" # PLEASE REVIEW -- Does it make sense to include this? Can modifying these break tests? (REMOVE COMMENT AFTER REVIEW)
+      # Workflows that can impact tests
+      - ".github/workflows/test*.yml"
+      # top-level files
+      - "package*.json"
+      - ".nvrmc" # Updates to node version can break tests
+      - "vite.*.ts" # vite.config.ts, vite.vitest.config.ts, vitest.workspace.ts
+      - "tsconfig*.json" # tsconfig.json tweaking can impact compilation
+      - "global.d.ts" # I think this can impact tests? (REMOVE COMMENT AFTER REVIEW)
+      - ".env.*" # PLEASE REVIEW -- can these impact test behavior?
+      # Blanket negations for files that cannot impact tests
+      - "!**/*.py" # No .py files
+      - "!**/*.sh" # No .sh files
+      - "!**/*.md" # No .md files
+      - "!**/*.png" # TODO: Can the deletion of these files break tests? (REMOVE COMMENT AFTER REVIEW)
+      - "!**/*.jpg" # TODO: Can the deletion of these files break tests? (REMOVE COMMENT AFTER REVIEW)
```

```suggestion
      - "!**/*.png"
      - "!**/*.jpg"
```
I don't think image files should impact tests.

---

Are there any tests that rely on the presence of specific files?

---

Afaik there are no tests that care about the existence of any image files (though maybe we should try deleting them and seeing if the tests still run?).